src/findAll.ts

```diff
@@ -1,7 +1,194 @@
 import { Connection } from "jsforce";
+import xml2js from "xml2js";
+import fetch from "node-fetch";
+import { promisify } from "util";
 
+import timeout from "./timeout";
 import { createSOQL, parseQuery, Query } from "./soql";
 
+const parseXML = promisify(xml2js.parseString);
+
+interface Job {
+  id: string;
+}
+
+interface Batch {
+  id: string;
+}
+
+interface BatchStatus {
+  apexProcessingTime: number;
+  apiActiveProcessingTime: number;
+  createdDate: string;
+  id: string;
+  jobId: string;
+  numberRecordsFailed: number;
+  numberRecordsProcessed: number;
+  state: string;
+  stateMessage: string | null;
+  systemModstamp: string;
+  totalProcessingTime: number;
+}
+
+interface Result {
+  id: string;
+}
+
+interface JobXML {
+  jobInfo: {
+    id: string[];
+  };
+}
+
+async function getBatchStatus(
+  accessToken: string,
+  baseUrl: string,
+  jobId: string,
+  batchId: string
+): Promise<BatchStatus> {
+  const response = await fetch(`${baseUrl}/job/${jobId}/batch/${batchId}`, {
+    headers: {
+      "X-SFDC-Session": accessToken
+    }
+  });
+
+  const result = await response.json();
+
+  return result;
+}
+
+async function batchCompletion(
+  accessToken: string,
+  baseUrl: string,
+  jobId: string,
+  batchId: string
+): Promise<void> {
+  let batchStatus = await getBatchStatus(accessToken, baseUrl, jobId, batchId);
```

Comment:
could just call in beginning of the loop so this line isn't repeated 
```suggestion
  let batchStatus = {};
```